### ISSUE TITLE
send_receive_timeout with value less than 5 causes DB::ParsingException: Unsigned type must not contain '-' symbol. (CANNOT_PARSE_NUMBER)

### ISSUE THREAD

**Issue by bnomis** [ClickHouse/clickhouse-connect#259]
This code:

```python
self._progress_interval = str(min(120000, (send_receive_timeout - 5) * 1000))
```
https://github.com/ClickHouse/clickhouse-connect/blob/15e88b2056d6f1cd53ffbc5094187c4de85db3ab/clickhouse_connect/driver/httpclient.py#L155C1-L155C90

Will cause `self._progress_interval` to be negative if `send_receive_timeout` is less than 5. 

This negative value is then passed to clickhouse via a http parameter e.g. `http_headers_progress_interval_ms=-1000`

Clickhouse then returns an error: `DB::ParsingException: Unsigned type must not contain '-' symbol. (CANNOT_PARSE_NUMBER)`

It could be fixed by:

```python
self._progress_interval = str(max(0, min(120000, (send_receive_timeout - 5) * 1000)))
```

Also the `-5` and `120000` may be better defined as settings and documented.

**Comment by genzgd** [comment]
You are correct, although there really isn't a valid use case for a send/receive timeout of less than 5 seconds.

**Comment by genzgd** [comment]
In addition the progress interval is really designed to keep the connection alive -- especially since the Python HTTP libraries can't actually read the intermediate progress headers.  There's no real benefit (or meaningful cost) to changing it and making it configurable again doesn't really have a use case.

**Comment by bnomis** [comment]
I suggest documenting the behavior so others don't run into this issue

**Comment by genzgd** [comment]
The `CANNOT_PARSE_NUMBER` error is fixed in 0.6.18.
